packages/core/src/stateUtils.ts

```diff
@@ -1563,15 +1567,25 @@ function resolveAndExecuteActionsWithContext(
           params: actionParams
         }
       });
-      resolvedAction(actionArgs, actionParams);
+      try {
+        resolvedAction(actionArgs, actionParams);
+      } catch (err) {
+        executingCustomAction = false;
+        // rethrow err
+        throw err;
+      }
```

Comment:
this would have the same effect
```suggestion
      } finally {
        executingCustomAction = false;
      }
```